.ci/scripts/dswx_compare_opera_pge.py

```diff
@@ -0,0 +1,223 @@
+import argparse
+import numpy as np
+import os
+from osgeo import gdal
+import sys
+
+COMPARE_DSWX_HLS_PRODUCTS_ERROR_TOLERANCE_ATOL = 1e-6
+COMPARE_DSWX_HLS_PRODUCTS_ERROR_TOLERANCE_RTOL = 1e-5
+
+DEFAULT_METADATA_EXCLUDE_LIST = ['PROCESSING_DATETIME', 'DEM_SOURCE', 'LANDCOVER_SOURCE', 'WORLDCOVER_SOURCE']
+PREFIX = ' ' * 7
+
+def _get_prefix_str(current_flag, flag_all_ok):
+    """ Return an updated cumulative flag status and an OK/FAIL string for the current flag
+
+    Parameters
+    ----------
+    current_flag: bool
+        Flag of the most recent comparison operation status which will be used to
+        determine the return string and update the flag_all_ok value
+    flag_all_ok: bool
+        Flag holding the cumulative flag status
+
+    Returns
+    -------
+    flag_all_ok: bool
+        Updated cumulative status flag based on current_flag input
+    prefix_str: str
+        Formatted string with OK or FAIL status based on current_flag input
+    """
+    flag_all_ok = flag_all_ok and current_flag
+    prefix_str = '[OK]   ' if current_flag else '[FAIL] '
+    return flag_all_ok, prefix_str
+
+
+def compare_dswx_hls_products(file_1, file_2, metadata_exclude_list=[]):
+    """
+    Compare DSWx-HLS products in various ways
+
+    Parameters
+    ----------
+    file_1: str
+        Input 1 filename including path
+    file_2: str
+        Input 2 filename including path
+    metadata_exclude_list: list
+        Metadata keys to ignore for purposes of determining comparison ok or fail status
+
+    Returns
+    -------
+    flag_all_ok: bool
+        Overall comparison status
+    """
+
+    if not os.path.isfile(file_1):
+        print(f'ERROR file not found: {file_1}')
+        return False
+
+    if not os.path.isfile(file_2):
+        print(f'ERROR file not found: {file_2}')
+        return False
+
+    print(f'input 1 is {file_1}')
+    print(f'input 2 is {file_2}')
+
+    flag_all_ok = True
+
+    # TODO: compare projections ds.GetProjection()
+    layer_gdal_dataset_1 = gdal.Open(file_1, gdal.GA_ReadOnly)
+    geotransform_1 = layer_gdal_dataset_1.GetGeoTransform()
+    metadata_1 = layer_gdal_dataset_1.GetMetadata()
+    nbands_1 = layer_gdal_dataset_1.RasterCount
+
+    layer_gdal_dataset_2 = gdal.Open(file_2, gdal.GA_ReadOnly)
+    geotransform_2 = layer_gdal_dataset_2.GetGeoTransform()
+    metadata_2 = layer_gdal_dataset_2.GetMetadata()
+    nbands_2 = layer_gdal_dataset_2.RasterCount
+
+    # compare number of bands
+    flag_same_nbands =  nbands_1 == nbands_2
+    flag_all_ok, flag_same_nbands_str = _get_prefix_str(flag_same_nbands, flag_all_ok)
+    prefix = PREFIX
+    print(f'{flag_same_nbands_str}Comparing number of bands')
```

Comment:
Add the prefix at the beginning of this line, to keep the first 7 characters of each line clear of anything other than a comparison status.

Or you can remove it.  If you put the error messages after the `"[FAIL] Band 1 - Water classification (WTR)"` line, you don't need this one.